### PR TITLE
Component/header

### PR DESCRIPTION
**Pull request by EstebanIndiveri**
## Summary

Added a custom Header with customIcons component and navigation, changed styles in screen DetailBook

#### Galaxy S (or similar)
<a href="https://ibb.co/72c9mv7">Screenshot-1620255111</a>

<a href="https://ibb.co/zZzDkzr">Screenshot-1620255122</a>

#### Pixel 2/3 (or similar)
<a href="https://ibb.co/QkYMCCV">Screenshot-1620255075</a>

<a href="https://ibb.co/GM4kpSC">Screenshot-1620255083</a>

#### Horizontal test
<a href="https://ibb.co/BC80hQ8">Screenshot-2021-05-05-19-55-00-019-com-wbooks</a>

#### iPhone SE (or similar)
<a href="https://ibb.co/tZHw3FF">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-05-at-19-50-37</a>

<a href="https://ibb.co/VHs370G">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-05-at-19-50-45</a>

#### iPhone 6/7/8 (or similar)
<a href="https://ibb.co/CJtYPwq">Simulator-Screen-Shot-i-Phone-8-2021-05-05-at-19-49-26</a><br /><a target='_blank' href='https://imgbb.com/'>speed jpg</a><br />

<a href="https://ibb.co/vXSJrVL">Simulator-Screen-Shot-i-Phone-8-2021-05-05-at-19-49-47</a>

#### iPhone 11/12 Base/Pro/Pro Max (or similar)
<a href="https://ibb.co/9rV4XF1">Simulator-Screen-Shot-i-Phone-11-2021-05-05-at-19-50-19</a>

<a href="https://ibb.co/Zgx6sCM">Simulator-Screen-Shot-i-Phone-11-2021-05-05-at-19-50-25</a>

## Trello Card

[Change!] Link to the associated Trello card.
[Header Trello Card](https://trello.com/c/RPvXQ03H/15-n-74-header)

### CODE REVIEW

File: wbooks/src/app/components/BackArrowIcon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import styles from './styles';
+
+export default function BackArrowIcon(): ReactElement {
+  const navigation = useNavigation();
+  const handlePress = () => {
+    navigation.goBack();
+  };
+  return (
+    <TouchableOpacity onPress={handlePress}>
+      <View style={styles.imageContainer}>
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/components/BackArrowIcon/index.tsx]: this view not is necesary, use styles in `TouchableOpacity`

File: wbooks/src/app/components/BackArrowIcon/styles.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  imageContainer: {
+    marginLeft: 10
+  },
+  image: {
+    height: 20,
+    width: 20
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/components/BackArrowIcon/styles.js]: the values is same, use conts, Example: `const ICON_SIZE = 20`

File: wbooks/src/app/components/Header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactElement } from 'react';
+import { View, Image } from 'react-native';
+import headerImage from '@assets/General/bc_nav_bar.png';
+
+import styles from './styles';
+
+export default function Header(): ReactElement {
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/components/Header/index.tsx]: @mattgle `ReactElement` is necesary?

Comment (mattgle) [reply wbooks/src/app/components/Header/index.tsx]: No is not needed. But is now wrong.

File: wbooks/src/app/components/BackArrowIcon/index.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import backArrow from '@assets/General/ic_back.png';
+import { useNavigation } from '@react-navigation/core';
+
+import styles from './styles';
+
+export default function BackArrowIcon(): ReactElement {
+  const navigation = useNavigation();
+  const handlePress = () => {
+    navigation.goBack();
+  };
```
Comment (mattgle) [inline wbooks/src/app/components/BackArrowIcon/index.tsx]:
```suggestion
  const handlePress = () =>     navigation.goBack();

```

File: wbooks/src/app/components/BackArrowIcon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React, { ReactElement } from 'react';
+import { Image, TouchableOpacity } from 'react-native';
+import backArrow from '@assets/General/ic_back.png';
+import { useNavigation } from '@react-navigation/core';
+
+import styles from './styles';
+
+export default function BackArrowIcon(): ReactElement {
```
Comment (mattgle) [inline wbooks/src/app/components/BackArrowIcon/index.tsx]: The ReactElement is not needed, it's not wrong, but its not needed.
```suggestion
export default function BackArrowIcon() {
```

File: wbooks/src/app/components/NotificationIcon/index.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactElement } from 'react';
+import { View, Image } from 'react-native';
+import notifications from '@assets/General/ic_notifications.png';
```
Comment (mattgle) [inline wbooks/src/app/components/NotificationIcon/index.tsx]:
```suggestion
import icNotification from '@assets/General/ic_notifications.png';
```

File: wbooks/src/app/components/NotificationIcon/index.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+import styles from './styles';
+
+export default function NotificationIcon(): ReactElement {
+  return (
+    <View style={styles.imageContainer}>
+      <Image source={notifications} style={styles.image} resizeMethod="scale" resizeMode="contain" />
```
Comment (mattgle) [inline wbooks/src/app/components/NotificationIcon/index.tsx]:
```suggestion
      <Image source={icNotification} style={styles.image} resizeMethod="scale" resizeMode="contain" />
```

File: wbooks/src/app/components/NotificationIcon/index.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactElement } from 'react';
+import { View, Image } from 'react-native';
+import notifications from '@assets/General/ic_notifications.png';
+
+import styles from './styles';
+
+export default function NotificationIcon(): ReactElement {
+  return (
+    <View style={styles.imageContainer}>
```
Comment (mattgle) [inline wbooks/src/app/components/NotificationIcon/index.tsx]: Are you sure you need this View? Cant you just put the styles at the image?

File: wbooks/src/app/components/SerchIcon/index.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactElement } from 'react';
+import { View, Image } from 'react-native';
+import searchIcon from '@assets/General/ic_search.png';
+
+import styles from './styles';
+
+export default function SeachIcon(): ReactElement {
```
Comment (mattgle) [inline wbooks/src/app/components/SerchIcon/index.tsx]: This component is exactly the same as the NotificationIcon, you should make a generic component and send the image as a prop.

**Review by mattgle — CHANGES_REQUESTED** [review]
Great work! Some comments!

**Review by mattgle — APPROVED** [review]

**Pull request merged**